bin/cml/runner/launch.js

```diff
@@ -588,6 +591,17 @@ exports.options = kebabcaseKeys({
     description: 'Specifies the subnet to use within AWS',
     alias: 'cloud-aws-subnet-id'
   },
+  cloudKubernetesNodeSelector: {
+    type: 'array',
+    string: true,
+    default: ['accelerator=infer'],
```

Comment:
> Can we keep the default value logic on the provided side?
> 
> I would prefer it match the other "provider" specific options and have a default value of `''` see `cloudAwsSubnet` as an example.

@dacbd, I tried to set the default value logic on the Terraform Provider Iterative side with no success.

I tried the following approaches:

```go
// https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-behaviors#default
"kubernetes_node_selector": &schema.Schema{
	Type:     schema.TypeMap,
	ForceNew: true,
	Optional: true,
	Elem: &schema.Schema{
		Type: schema.TypeString,
	},
	Default: map[string]string{
		"accelerator": "infer",
	},
},
```

and

```go
// https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-behaviors#defaultfunc
"kubernetes_node_selector": &schema.Schema{
	Type:     schema.TypeMap,
	ForceNew: true,
	Optional: true,
	Elem: &schema.Schema{
		Type: schema.TypeString,
	},
	DefaultFunc: func() (any, error) {
		return map[string]string{
			"accelerator": "infer",
		}, nil
	},
},
```

but every time I wasn't specifying the `--cloud-kubernetes-node-selector` argument to CML, the default value were just ignored on the provider side.

Have I missed something regarding this aspect?

For the moment, I left the default value on the CML side to be more explicit to the end user that the inference is made on the `accelerator` key.

Reply:
Sure, I mostly don't want to have yargs populate a default value. With how you had it, it would populate into the generated terraform even when it's not used, which for others, down the line, may cause confusion. (for example, why is there a k8s node selector when I'm using AWS).

Reply:
You can utilize this line to see what I mean: https://github.com/iterative/cml/blob/1be24edaa817de320a657ec3ad1182e145aecef7/src/terraform.js#L123

Reply:
with this line: i believe we are safe to remove the default value from the cml side: https://github.com/iterative/terraform-provider-iterative/pull/747/files#diff-08560ef5459da3ede56dd9700b81f84384dbff1de69918854a2b8e2da0155155R72